agent/agent.go

```diff
@@ -124,82 +119,60 @@ func NewAgent(logger *log.Logger, config *Config) (*Agent, error) {
-		logrus.SetOutput(os.Stdout)
-
-		// Only log the warning severity or above.
-		logrus.SetLevel(logrus.WarnLevel)
-
-		proxy := toxiproxy.NewProxy()
-		proxy.Name = "proxy"
-		proxy.Listen = config.ProxyAddr.String()
-		proxy.Upstream = config.Addr.String()
-
-		sidecar := toxiproxy.NewServer()
-		if err := sidecar.Collection.Add(proxy, true); err != nil {
-			// In 5% of the cases not sure why but this port is already in use, then, remove the proxy
-			// and use the normal connection.
-			// I think this should work fine
-			config.ProxyAddr = nil
-			logger.Printf("[INFO]: Fallback proxy")
-		} else {
-			go func() {
-				// pick up every link created in the proxy and add the specific latency toxic
-				linkCh := make(chan *toxiproxy.ToxicLink, 1000)
-				proxy.Toxics.LinkCh = linkCh
+	logger.Printf("[INFO]: Address: %s", config.Addr.String())
+
+	logrus.SetFormatter(&logrus.JSONFormatter{})
 
-				for {
-					link := <-linkCh
+	// Output to stdout instead of the default stderr
+	// Can be any io.Writer, see below for File example
+	logrus.SetOutput(os.Stdout)
+
+	// Only log the warning severity or above.
+	logrus.SetLevel(logrus.WarnLevel)
+
+	// In 5% of the cases not sure why but this port is already in use, then, remove the proxy
+	// and use the normal connection.
+	// I think this should work fine
+
+	go func() {
+		// pick up every link created in the proxy and add the specific latency toxic
+		linkCh := make(chan string, 1000)
 
-					go func(link *toxiproxy.ToxicLink) {
+		for {
+			link := <-linkCh
 
-						name := link.Name()
-						name = strings.TrimSuffix(name, "upstream")
-						name = strings.TrimSuffix(name, "downstream")
+			go func() {
 
-						// fmt.Println(name)
+				name := link
+				name = strings.TrimSuffix(name, "upstream")
+				name = strings.TrimSuffix(name, "downstream")
 
-						// ping that ip in name on the http port to get the name of the destination city
+				// fmt.Println(name)
 
-						// the ip we get from name is the bind-address which is 300x, the one from the http port is 400x
-						ip := strings.Replace(name, ":30", ":40", -1)
+				// ping that ip in name on the http port to get the name of the destination city
 
-						var latencyF int64
+				// the ip we get from name is the bind-address which is 300x, the one from the http port is 400x
+				ip := strings.Replace(name, ":30", ":40", -1)
 
-						dest, err := query("http://" + ip + "/system/city")
-						if err != nil {
-							logger.Printf("[INFO]: Add latency: name=%s, ip=%s default", name, ip)
-							latencyF = time.Duration(100 * time.Millisecond).Milliseconds()
-						} else {
-							proxyLatency := latency.FindLatency(config.City, dest)
+				var latencyF int64
 
-							logger.Printf("[INFO]: Add latency: name=%s, ip=%s, city=%s, latency=%s", name, ip, dest, proxyLatency)
-							// logger.Printf("[INFO]: Add latency: ip=%s", name)
+				dest, err := query("http://" + ip + "/system/city")
```

Comment:
I do not think we will need this either, as per the version on my repo there should be some generic `transport` entity that creates all the latency `Transport` and that one will keep track of which city does each instance belongs to.

Reply:
ok, removed